### ISSUE TITLE
Errors Occurring After Months of Successful Use

### ISSUE THREAD

**Issue by silverhero7** [briansrls/Robinhood-for-Google-Finance#12]
Greetings!
I've been using this amazing exporter for quite some time. Admittedly, I'm not very advanced in coding whatsoever. I had installed and used your code for months and all of a sudden started getting the error below. Later I tried installing Python3 and trying again with no luck. Any help is greatly appreciated!
Thanks, Nick

Traceback (most recent call last):
  File "gf-export.py", line 33, in <module>
    logged_in = robinhood.login(username=username, password=password)
  File "/Users/Nick/Documents/Robinhood/Robinhood.py", line 88, in login
    res = res.json()
  File "/Users/Nick/Library/Python/2.7/lib/python/site-packages/requests/models.py", line 896, in json
    return complexjson.loads(self.text, **kwargs)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/decoder.py", line 384, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
MacBook-Pro:robinhood Nick$

**Comment by briansrls** [comment]
Hello,

Thank you for the kind words. I see the error you have, I’ll get back to
you with what I think the cause may be.

Brian

**Comment by silverhero7** [comment]
Thank you so much!

**Comment by briansrls** [comment]
Sorry for such a long delay, I've had little time to look at this in my free time.

I believe the issue is stemming from here: 
https://github.com/Jamonek/Robinhood/issues/137
And here is some more discussion:
https://github.com/Jamonek/Robinhood/issues/135
https://github.com/Jamonek/Robinhood/issues/134

You can actually see the same error you get in the second link.

Essentially, robinhood appears to have changed their external API a bit and they also appear to not be keen on individuals using it in this way (referencing issue 137 linked above, support actually forced this user to submit govt ID and told them to never use the API going forward or risk account termination!).  

We can probably continue using the API for a bit longer.  There are fixes posted in the issues I linked before that I will *hopefully* be trying this weekend if time permits.

I would expect this to get more difficult as time goes on.  My worry is that users may be banned/accounts closed.  Obviously we don't want to risk this no matter how useful these tools are.  

TLDR; Please use these tools at your own risk.  When things concerning SEC/FINRA get involved robinhood would not be interested in dealing with the hassle.

**Comment by silverhero7** [comment]
Thank you so much for looking I to this! I sincerely appreciate your effort and expertise. I think I’ll play it safe for now and avoid using the API. 

Best,
Nick

**Comment by briansrls** [comment]
Understood Nick, I think that is a wise decision.  I will probably have to include a warning in the script so that before a user logs in, it will notify them that this could be dangerous and maybe link them to the support response.

I will continue to look into the issue regardless.

**Comment by arh** [comment]
How can robinhood support tell that the api call is originated from this library or their iOS app? Is there any way to make it exactly the same?

**Comment by wezzybytes** [comment]
@arh I am sure the iOS/Android Robinhood app adds additional parameters to each requests it generates. Since the app is closed-source, we can't know for sure which parameters. Requests are probably signed with their SSL certs too, so can't even sniff it with Wireshark.

**Comment by briansrls** [comment]
I've updloaded a potential fix for these issues, thanks to the robinhood repo.  I've kept them in my dev branch for now.  https://github.com/briansrls/Robinhood-for-Google-Finance/pull/14

However, 2fa is not working right now.  Use at your own risk!

**Comment by briansrls** [comment]
I've pushed my changes to master.  Please try re-pulling the latest changes and let me know if you still have any issues.
